examples/platforms/cc2538/uart.c

```diff
@@ -121,24 +129,43 @@ void processReceive(void)
 {
     uint16_t remaining;
 
-    VerifyOrExit(sReceiveLength > 0, ;);
+    // Copy static state to local variable to prevent multiple reads
+    RecvBuffer recv = sReceive;
+
+    // Bail out if we have no data to process
+    VerifyOrExit(recv.Length > 0, ;);
 
-    remaining = kReceiveBufferSize - sReceiveHead;
+    // Calculate the remaining buffer before wrap around
+    remaining = kReceiveBufferSize - recv.Offset;
 
-    if (sReceiveLength >= remaining)
+    // If the data wraps around, process the first part
+    if (recv.Length >= remaining)
     {
-        otPlatUartReceived(sReceiveBuffer + sReceiveHead, remaining);
-        sReceiveHead = 0;
-        sReceiveLength -= remaining;
+        otPlatUartReceived(sReceiveBuffer + recv.Offset, remaining);
+
+        // Reset the buffer offset to start and set remaining
+        recv.Offset = 0;
+        remaining = recv.Length - remaining;
+    }
+    else
+    {
+        // Since there is no wrap around, the entire length is remaining
+        remaining = recv.Length;
     }
 
-    if (sReceiveLength > 0)
+    // For any data remaining, process it
+    if (remaining > 0)
     {
-        otPlatUartReceived(sReceiveBuffer + sReceiveHead, sReceiveLength);
-        sReceiveHead += sReceiveLength;
-        sReceiveLength = 0;
+        otPlatUartReceived(sReceiveBuffer + recv.Offset, remaining);
+        recv.Offset += remaining;
```

Comment:
I believe this should be `recv.Offset = (recv.Offset + remaining) % kReceiveBufferSize;` but it isn't necessary, because line 167 will take care of it.